src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,170 @@
+import type { Options } from './index';
+
+const boolOpts = [
+  'makeCumulative',
+  'loop',
+  'showIcons',
+  'showGroups',
+  'mouseControls',
+  'keyboardControls',
+  'autorun',
+  'injectStyles',
+  'highlightBars',
+  'selectBars',
+  'fixedScale',
+] as const;
+const numberOpts = [
+  'labelsWidth',
+  'tickDuration',
+  'topN',
+  'minHeight',
+  'minWidth',
+  'marginTop',
+  'marginRight',
+  'marginBottom',
+  'marginLeft',
+] as const;
+const strOpts = ['theme', 'startDate', 'endDate'] as const;
+const strOrNumberOpts = ['colorSeed', 'inputHeight', 'inputWidth', 'height', 'width'] as const;
+const strOrFuncOpts = ['title', 'subTitle', 'dateCounter', 'caption'] as const;
+
+const validDataShapes = ['long', 'wide'];
+const validDataTypes = ['json', 'csv', 'tsv', 'xml'];
+const validLabelsPositions = ['inside', 'outside'];
+const validControlButtons = ['all', 'play', 'none'];
+const validOverlays = ['all', 'play', 'repeat', 'none'];
+const validFillDateGapsIntervals = [null, 'year', 'month', 'day'];
+const validFillDateGapsValues = ['last', 'interpolate'];
+
+export function validateOptions(options: Partial<Options>): Partial<Options> {
+  const newOptions: Partial<Options> = {};
+
+  // Validate boolean options
+  boolOpts.forEach((opt) => {
+    if (!is(options[opt], 'boolean')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate number options
+  numberOpts.forEach((opt) => {
+    if (!is(options[opt], 'number')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string options
+  strOpts.forEach((opt) => {
+    if (!is(options[opt], 'string')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string or number options
+  strOrNumberOpts.forEach((opt) => {
```

Comment:
what happens if we do this?
```ts
{ height: 'hello' }
```

We should check that the string can indeed be converted to a number
```ts
// Validate string or number options
strOrNumberOpts.forEach((opt) => {
  if (!is(options[opt], 'string') && !is(options[opt], 'number')) return;
  if (is(options[opt], 'string') && isNaN(Number(options[opt]))) return;
  newOptions[opt] = options[opt];
});
```

Reply:
'Is' function handles this situation

```
function is(value: any, type: types, arrayType?: types): boolean {
  if (typeof value === 'undefined') return false;

  if (type === 'number') {
    return typeof value === 'number' && !isNaN(Number(value));
  }

  if (type === 'boolean') {
    return typeof value === 'boolean' || value === 'true' || value === 'false';
  }

  if (type === 'array') {
    if (!Array.isArray(value)) return false;
    if (!arrayType) return true;

    return value.every((val) => is(val, arrayType));
  }

  if (type === 'object') {
    return value !== null && typeof value === 'object';
  }

  return typeof value === type;
}
```